addons/knowledge/models/knowledge_article.py

```diff
@@ -748,29 +758,47 @@ def get_field_info(article, field_name):
     # Permission and members handling methods
     # ---------------------------------------
 
-    def set_article_permission(self, permission):
+    def set_internal_permission(self, permission):
+        """
+        :param permission (str): permission ('none', 'read' or 'write')
+        """
         self.ensure_one()
-        if self.user_can_write:
-            self.write({'internal_permission': permission})
+        if not self.user_can_write:
+            return False
+        return self.write({'internal_permission': permission})
 
-    def set_member_permission(self, partner_id, permission):
+    def set_member_permission(self, member_id, permission):
+        """
+        :param member_id (int): member id
+        :param permission (str): permission ('none', 'read' or 'write')
+        """
         self.ensure_one()
-        if self.user_can_write:
-            member = self.sudo().article_member_ids.filtered(lambda member: member.partner_id.id == partner_id)
-            member.write({'permission': permission})
+        if not self.user_can_write:
+            return False
+        member = self.article_member_ids.filtered(lambda member: member.id == member_id)
+        return member.write({'permission': permission})
 
-    # def remove_member(self, partner_id):
-    #     self.ensure_one()
-    #     if self.user_can_write:
-    #         member = self.main_article_id.sudo().article_member_ids.filtered(lambda member: member.partner_id.id == partner_id)
-    #         member.unlink()
+    def remove_member(self, member_id):
+        """
+        :param member_id (int): member id
+        """
+        self.ensure_one()
+        member = self.main_article_id.article_member_ids.filtered(lambda member: member.id == member_id)
+        if not self.user_can_write and self.env.user.partner_id != member.mapped('partner_id'):
+            return False
+        return member.unlink()
 
     def invite_member(self, access_rule, partner_id=False, email=False, send_mail=True):
```

Comment:
You can make this method multi, taking only partner_ids and send_mail as param. 
As the caller always calls this with partner, never with email.